apps/ogr2ogr_lib.cpp

```diff
@@ -4400,6 +4401,48 @@ std::unique_ptr<TargetLayerInfo> SetupTargetLayer::Setup(OGRLayer* poSrcLayer,
         psInfo->m_oMapDomainToKV[poDomain] = std::move(oMapCodeValue);
     }
 
+    // Detect if we can directly passed the source feature to the CreateFeature()
+    // method of the target layer, without doing any copying of field content.
+    psInfo->m_bCanAvoidSetFrom = false;
+    if( !m_bExplodeCollections && iSrcZField == -1 )
```

Comment:
I might be missing something (the diff is really hard to read here unfortunately), but don't we also need to consider other settings like m_poClipSrc ?

Reply:
Good point. I now remember I was a bit perplexed by this because there is also m_poClipSrc being taken into account later in https://github.com/OSGeo/gdal/blob/5427fe95881c19adc67e8887f7d74efe86788e0c/apps/ogr2ogr_lib.cpp#L4983. It looked like the logic at https://github.com/OSGeo/gdal/blob/5427fe95881c19adc67e8887f7d74efe86788e0c/apps/ogr2ogr_lib.cpp#L4829 might be redundant, but I now believe that I understand it is for the case where someone would use -clipsrc to filter a geometry to and produce a a-spatial layer. But this situation will cause nSrcGeomFieldCount (>=1) != nDstGeomFieldCount (=0), which causes m_bCanAvoidSetFrom to be set to false. So I believe we're good as it.